internal/app/machined/pkg/controllers/seccomp/seccomp.go

```diff
@@ -0,0 +1,169 @@
+	if err := r.UpdateInputs([]controller.Input{
+		{
+			Namespace: runtimeres.NamespaceName,
+			Type:      runtimeres.MountStatusType,
+			ID:        pointer.To(constants.EphemeralPartitionLabel),
+			Kind:      controller.InputWeak,
+		},
+	}); err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-r.EventCh():
+		}
+
+		_, err := r.Get(ctx, resource.NewMetadata(runtimeres.NamespaceName, runtimeres.MountStatusType, constants.EphemeralPartitionLabel, resource.VersionUndefined))
+		if err != nil {
+			if state.IsNotFoundError(err) {
+				// in container mode EPHEMERAL is always mounted
+				if ctrl.V1Alpha1Mode != runtimetalos.ModeContainer {
+					// wait for the EPHEMERAL to be mounted
+					continue
+				}
+			} else {
+				return fmt.Errorf("error getting ephemeral mount status: %w", err)
+			}
+		}
+
+		break
+	}
+
+	// normal reconcile loop
+	if err := r.UpdateInputs([]controller.Input{
+		{
+			Namespace: config.NamespaceName,
+			Type:      config.MachineConfigType,
+			ID:        pointer.To(config.V1Alpha1ID),
+			Kind:      controller.InputWeak,
+		},
+	}); err != nil {
+		return err
+	}
+
+	r.QueueReconcile()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-r.EventCh():
+		}
+
+		cfg, err := r.Get(ctx, resource.NewMetadata(config.NamespaceName, config.MachineConfigType, config.V1Alpha1ID, resource.VersionUndefined))
+		if err != nil {
+			if state.IsNotFoundError(err) {
+				continue
+			}
+
+			return fmt.Errorf("error getting config: %w", err)
+		}
+
+		cfgProvider := cfg.(*config.MachineConfig).Config()
+
+		touchedIDs := map[string]struct{}{}
+
+		for _, profile := range cfgProvider.Machine().SeccompProfiles() {
+			value, errJSON := json.Marshal(profile.Value())
+			if errJSON != nil {
+				return fmt.Errorf("error marshaling seccomp profile: %w", errJSON)
+			}
+
+			if err = r.Modify(ctx, seccomp.NewSeccompProfile(profile.Name()), func(res resource.Resource) error {
+				*res.(*seccomp.Profile).TypedSpec() = seccomp.ProfileSpec{
+					Name:  profile.Name(),
+					Value: profile.Value(),
+				}
+
+				return nil
+			}); err != nil {
+				return err
+			}
+
+			touchedIDs[profile.Name()] = struct{}{}
+
+			if err = ioutil.WriteFile(filepath.Join(ctrl.SeccompProfileDirectory, profile.Name()), value, 0o644); err != nil {
+				return fmt.Errorf("error writing seccomp profile: %w", err)
+			}
+		}
+
+		// list keys for cleanup
+		list, err := r.List(ctx, resource.NewMetadata(seccomp.NamespaceName, seccomp.ProfileType, "", resource.VersionUndefined))
+		if err != nil {
+			return fmt.Errorf("error listing seccomp profiles: %w", err)
+		}
+
+		for _, res := range list.Items {
+			if _, ok := touchedIDs[res.Metadata().ID()]; !ok {
+				if err := r.Destroy(ctx, res.Metadata()); err != nil {
+					return fmt.Errorf("error deleting seccomp profile: %w", err)
+				}
+
+				if err := os.Remove(filepath.Join(ctrl.SeccompProfileDirectory, res.Metadata().ID())); err != nil {
```

Comment:
we should probably ignore `not found` here to handle cases when the resource was created, but the file wasn't

Reply:
also, as `EPHEMERAL` persists across reboots, but resources do not persist, I think it makes sense to explicitly clean up seccomp profiles in this directory which we don't have in the resources.

Reply:
that's a good point, I'll also test that wiping ephemeral re-creates the files on next boot.

Reply:
tested and verfied